### PR TITLE
cephfs-shell: getxattr fail while the xattr's length > 256

### PR DESCRIPTION
**Pull request by tengjie5**
Fixes:  https://tracker.ceph.com/issues/62545
Signed-off-by: teng jie <tengjie5@asiainfo.com>
 Please enter the commit message for your changes. Lines starting

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/tools/cephfs/shell/cephfs-shell [hunk 1]
```diff
@@ -1633,7 +1633,7 @@ class CephFSShell(Cmd):
         """
         try:
             poutput('{}'.format(cephfs.getxattr(args.path,
-                                to_bytes(args.name)).decode('utf-8')))
+                                to_bytes(args.name), size=65536).decode('utf-8')))
```
Comment (mchangir) [inline src/tools/cephfs/shell/cephfs-shell:1636]: Is this a problem with the xattr name or value ?
Please mention the same as part of the commit message as the description in the tracker is confusing too.

Comment (tengjie5) [reply src/tools/cephfs/shell/cephfs-shell:1636]: The problem is about length of xattr's value.when the value's length >256, report this  error.I've read code of function getxattr，default value of size is 256，so I set it to 65536 like linux.

Comment (neesingh-rh) [reply src/tools/cephfs/shell/cephfs-shell:1636]: Linux uses MAX_SIZE as power 16, seems ok to me.

**Review by neesingh-rh — APPROVED** [review]

File: src/tools/cephfs/shell/cephfs-shell (hunk 1 above)
Comment (mchangir) [reply src/tools/cephfs/shell/cephfs-shell:1636]: Please rename the commit title to:
**cephfs-shell: bump up acceptable xattr value len to 64K**

Also, remove the spurious line in the commit message:
`Please enter the commit message for your changes. Lines starting`

Comment (tengjie5) [reply src/tools/cephfs/shell/cephfs-shell:1636]: @mchangir  Title has been modified，thanks for the review.

**Comment by tengjie5** [comment]
@neesingh-rh  hi，excuse me，Is there a problem with the API test？it reported error as follows，i don't think it caused by my  pr.

AssertionError: timeout expired in wait_for_all_osds_up
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 1537, in 
    exec_test()
  File "/home/jenkins-build/build/workspace/ceph-api/build/../qa/tasks/vstart_runner.py", line 1392, in exec_test
    LocalCephCluster(LocalContext()).mon_manager.wait_for_all_osds_up(timeout=30)
  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/ceph_manager.py", line 2778, in wait_for_all_osds_up
    assert time.time() - start < timeout, \
AssertionError: timeout expired in wait_for_all_osds_up

**Comment by joscollin** [comment]
jenkins test api

**Comment by tengjie5** [comment]
> jenkins test api

@joscollin  api test has a new error，could you help me to take a look? thanks.

**Comment by joscollin** [comment]
> > jenkins test api
> 
> @joscollin api test has a new error，could you help me to take a look? thanks.

Could you please rebase your branch and force push again?
(git pull and force push)

**Comment by tengjie5** [comment]
> > > jenkins test api
> > 
> > 
> > @joscollin api test has a new error，could you help me to take a look? thanks.
> 
> Could you please rebase your branch and force push again? (git pull and force push)

It's ok now, thanks.

File: src/tools/cephfs/shell/cephfs-shell (hunk 1 above)
Comment (rishabh-d-dave) [reply src/tools/cephfs/shell/cephfs-shell:1636]: > The problem is about length of xattr's value.when the value's length >256, report this error.I've read code of function getxattr，default value of size is 256，so I set it to 65536 like linux.

I think this detail should be added to the commit message and to the tracker ticket.

**Review by joscollin — APPROVED** [review]

**Review by kotreshhr — COMMENTED** [review]
Could we add a testcase for this in `qa/tasks/cephfs/test_cephfs_shell.py`?

**Review by rishabh-d-dave — APPROVED** [review]
CephFS integration tests ran successfully - https://tracker.ceph.com/projects/cephfs/wiki/Main#3-Nov-2023

**Pull request merged**